common.js

```diff
@@ -62,3 +65,30 @@ export function win2nix(path) {
   }
   return path.replace(/\\/g, '/').replace(/ /g, '\\ ')
 }
+
+export function setupPath(newPathEntries) {
+  const envPath = isWin ? 'Path' : 'PATH'
+  const originalPath = process.env[envPath].split(path.delimiter)
+  let cleanPath = originalPath.filter(entry => !/\bruby\b/i.test(entry))
+
+  if (cleanPath.length !== originalPath.length) {
+    core.startGroup(`Cleaning ${envPath}`)
+    console.log(`Entries removed from ${envPath} to avoid conflicts with Ruby:`)
+    for (const entry of originalPath) {
+      if (!cleanPath.includes(entry)) {
+        console.log(`  ${entry}`)
+      }
+    }
+    core.exportVariable(envPath, cleanPath.join(path.delimiter))
+    core.endGroup()
+  }
+  let newPath
+  if (isWin) {
+    // add MSYS2 path to all for bash shell
+    const msys2 = ['C:\\msys64\\mingw64\\bin', 'C:\\msys64\\usr\\bin']
+    newPath = [...newPathEntries, ...msys2]
+  } else {
+    newPath = newPathEntries
+  }
+  core.addPath(newPath.join(path.delimiter))
```

Comment:
A loop over paths with a ` core.addPath()` seems nicer (but would need to be done in reverse order).
Since this work though it seems fine.